modules/tinymce/src/core/test/ts/browser/content/EditorContentTest.ts

```diff
@@ -94,6 +94,46 @@ describe('browser.tinymce.core.content.EditorContentTest', () => {
     assertEventsContentType();
   });
 
+  it('TINY-8578: getContent text, empty line in div', () => {
+    const editor = hook.editor();
+    editor.setContent('<div><p></p></div>');
+    const text = editor.getContent({ format: 'text' });
+    const expected = '';
+    Assertions.assertEq('Should be expected text', expected, text);
+    assertEventsFiredInOrder();
+    assertEventsContentType();
+  });
+
+  it('TINY-8578: getContent text, empty line', () => {
+    const editor = hook.editor();
+    editor.setContent('<p></p>');
+    const text = editor.getContent({ format: 'text' });
+    const expected = '';
+    Assertions.assertEq('Should be expected text', expected, text);
+    assertEventsFiredInOrder();
+    assertEventsContentType();
+  });
+
+  it('TINY-8578: getContent text, two empty lines in div', () => {
+    const editor = hook.editor();
+    editor.setContent('<div><p></p><p></p></div>');
+    const text = editor.getContent({ format: 'text' });
+    const isSafari = PlatformDetection.detect().browser.isSafari();
+    Assertions.assertEq('Should be expected text', isSafari ? '\n\n' : '\n\n\n\n', text);
```

Comment:
Yeah, we'd need to make something quite big and complex to work around it. I'd say it's something we should look into with time. But for now we probably don't have any option than to let this issue be browser dependent, I don't think we want to ad-hoc a parser of that complexity. At least not without deciding what output is correct.

Reply:
We did attempt generic "get text" logic in TBIO. It only extracts the text from text nodes, everything else it decides how many newlines to add. The logic was included in the port of libraries to TinyMCE:
https://github.com/tinymce/tinymce/blob/2d8b95517a43d1f1967a2f3898371865a2fa18e4/modules/phoenix/src/main/ts/ephox/phoenix/api/dom/DomExtract.ts#L31-L33
Which ends up in this code:
https://github.com/tinymce/tinymce/blob/2d8b95517a43d1f1967a2f3898371865a2fa18e4/modules/phoenix/src/main/ts/ephox/phoenix/extract/ExtractText.ts#L13-L17

I'm not saying we need to use this _now_, but it might be worth logging as a future JIRA improvement.